src/throttler/throttler.ts

```diff
@@ -140,17 +185,50 @@ export abstract class Throttler<T, R> {
     this.handle(this.nextWaitingTaskIndex());
   }
 
-  async handle(cursorIndex: number): Promise<void> {
+  stats(): ThrottlerStats {
+    return {
+      max: this.max,
+      min: this.min,
+      avg: this.avg,
+      active: this.active,
+      complete: this.complete,
+      success: this.success,
+      errored: this.errored,
+      retried: this.retried,
+      total: this.total,
+      elapsed: Date.now() - this.startTime,
+    };
+  }
+
+  taskName(cursorIndex: number): string {
     const taskData = this.taskDataMap.get(cursorIndex);
     if (!taskData) {
-      throw new Error(`taskData.get(${cursorIndex}) does not exist`);
+      return "finished task";
     }
-    const task = taskData.task;
-    const tname = this.taskName(cursorIndex);
-    const t0 = Date.now();
+    return typeof taskData.task === "string" ? taskData.task : `index ${cursorIndex}`;
+  }
+
+  private initializeTask(taskData: TaskData<T, R>, cursorIndex: number): void {
+    this.taskDataMap.set(cursorIndex, {
+      ...taskData,
+      status: TaskStatus.RUNNING,
+    });
+    if (taskData.timeout) {
+      this.taskTimeoutIdMap.set(
+        cursorIndex,
+        setTimeout(() => {
+          this.isTimedOutMap.set(cursorIndex, true);
```

Comment:
Instead of tracking this bit of information in a separate map, I think we can track it in `TaskData` itself (in the `taskDataMap`).

Reply:
Added `isTimedOut: boolean` prop in `TaskData`